File: tourneyAPI/Models/Entities/Characters/LittleMac.cs
namespace Entities;

using Enums;


// Defines the competitive profile metadata for this playable character.
public class LittleMac : Character
{
    // Initializes this character's default competitive attributes for matchmaking and tier logic.
    public LittleMac()
    {
        Id = CharacterId.Link;
        characterName = CharacterName.LITTLE_MAC;
        archetype = Archetype.BAIT_AND_PUNISH;
        fallSpeed = FallSpeed.FAST_FALLERS;
        weightClass = WeightClass.BALLOONWEIGHT;
        tierPlacement = TierPlacement.E;
    }

}

File: tourneyClient/src/models/entities/Characters/LittleMac.ts
import { CharacterId } from "@/models/Enums/CharacterId";
import { Character } from "../Character"
import { CharacterName } from "@/models/Enums/CharacterName";
import { Archetype } from "@/models/Enums/Archetype";
import { FallSpeed } from "@/models/Enums/FallSpeed";
import { WeightClass } from "@/models/Enums/WeightClass";
import { TierPlacement } from "@/models/Enums/TierPlacement";

export const LittleMac: Character =
{
    id: CharacterId.LittleMac,
    characterName: CharacterName.LITTLE_MAC,
    archetype: Archetype.BAIT_AND_PUNISH,
    fallSpeed: FallSpeed.FAST_FALLERS,
    weightClass: WeightClass.BALLOONWEIGHT,
    tierPlacement: TierPlacement.E
}

export default LittleMac; 